From 9412c019efb0b7d389ef2ea2851c9a9b5fddcb57 Mon Sep 17 00:00:00 2001
From: Rabeeh Khoury <rabeeh@solid-run.com>
Date: Fri, 6 Dec 2013 17:18:39 +0200
Subject: [PATCH 18/20] Tune mmdc frequency changes to microsom

Enable DDR ODT 120ohm; in the future this can potentially go away and thus making the system consume less power (about 100mA from the 5V rail on max load)

Signed-off-by: Rabeeh Khoury <rabeeh@solid-run.com>
---
 arch/arm/mach-mx6/mx6_mmdc.c |   24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/arch/arm/mach-mx6/mx6_mmdc.c b/arch/arm/mach-mx6/mx6_mmdc.c
index f0771b6..ea9e299 100644
--- a/arch/arm/mach-mx6/mx6_mmdc.c
+++ b/arch/arm/mach-mx6/mx6_mmdc.c
@@ -70,6 +70,17 @@ static int curr_ddr_rate;
 #define MIN_DLL_ON_FREQ		333000000
 #define MAX_DLL_OFF_FREQ		125000000
 
+#ifdef CONFIG_MACH_MICROSOM
+unsigned long ddr3_dll_mx6q[][2] = {
+	{0x0c, 0x0},
+	{0x10, 0x0},
+	{0x1C, 0x02088032},
+	{0x1C, 0x0208803a},
+	{0x1C, 0x09408030},
+	{0x1C, 0x09408038},
+	{0x818, 0x7},
+};
+#else
 unsigned long ddr3_dll_mx6q[][2] = {
 	{0x0c, 0x0},
 	{0x10, 0x0},
@@ -79,6 +90,7 @@ unsigned long ddr3_dll_mx6q[][2] = {
 	{0x1C, 0x08408038},
 	{0x818, 0x0},
 };
+#endif
 
 unsigned long ddr3_calibration[][2] = {
 	{0x83c, 0x0},
@@ -91,6 +103,17 @@ unsigned long ddr3_calibration[][2] = {
 	{0x4850, 0x0},
 };
 
+#ifdef CONFIG_MACH_MICROSOM
+unsigned long ddr3_dll_mx6dl[][2] = {
+	{0x0c, 0x0},
+	{0x10, 0x0},
+	{0x1C, 0x00008032},
+	{0x1C, 0x0000803a},
+	{0x1C, 0x05208030},
+	{0x1C, 0x05208038},
+	{0x818, 0x0},
+};
+#else
 unsigned long ddr3_dll_mx6dl[][2] = {
 	{0x0c, 0x0},
 	{0x10, 0x0},
@@ -100,6 +123,7 @@ unsigned long ddr3_dll_mx6dl[][2] = {
 	{0x1C, 0x07208038},
 	{0x818, 0x0},
 };
+#endif
 
 unsigned long iomux_offsets_mx6q[][2] = {
 	{0x5A8, 0x0},
-- 
1.7.9.5

